src/metabase/api/card.clj

```diff
@@ -435,20 +436,24 @@
   with queries they wouldn't be allowed to run!"
   [query]
   {:pre [(map? query)]}
-  (when-not (query-perms/can-run-query? query)
-    (let [required-perms (try
-                           (query-perms/required-perms query :throw-exceptions? true)
-                           (catch Throwable e
-                             e))]
-      (throw (ex-info (tru "You cannot save this Question because you do not have permissions to run its query.")
-                      {:status-code    403
-                       :query          query
-                       :required-perms (if (instance? Throwable required-perms)
-                                         :error
-                                         required-perms)
-                       :actual-perms   @api/*current-user-permissions-set*}
-                      (when (instance? Throwable required-perms)
-                        required-perms))))))
+  (let [preprocessed-query (try (some-> query qp/preprocess)
+                                (catch Exception e
+                                  (log/warn e "Error preprocessing query for permissions check, using original!")
```

Comment:
should this be a warn? (i genuinely don't know. Want us to justify this level, not saying we have to change it)

Reply:
You'd think that if you can't preprocess a query you can't run it, which is what we're trying to check here anyway. So maybe we should just let the error bubble up? But maybe that feels a little risky.

Reply:
When I added this I was thinking that the only reason this would fail was if the query was malformed somehow, which was unlikely to happen in production (since it's our own frontend constructing it) and a problem worthy of a warning if it did.

But I think this could happen for other reasons which might be caused by user error. So I've changed this to `log/info` instead.